packages/page-claims/src/index.tsx

```diff
@@ -73,183 +62,155 @@ const Signature = styled.textarea`
-                <Signature
-                  onChange={this.onChangeSignature}
-                  placeholder={`{\n  "address": "0x ...",\n  "msg": "${prefix}:...",\n  "sig": "0x ...",\n  "version": "2"\n}`}
-                  rows={10}
-                />
-                {(step === Step.Sign) && (
-                  <Button.Group>
-                    <Button
-                      icon='sign-in'
-                      isDisabled={!accountId || !signature}
-                      label={t('Confirm claim')}
-                      onClick={this.setStep(Step.Claim)}
-                    />
-                  </Button.Group>
-                )}
-              </Card>
-            )}
-          </Column>
-          <Column showEmptyText={false}>
-            {(step >= Step.Claim) && (
-              <ClaimDisplay
-                button={this.renderTxButton()}
+          )}
+        </Column>
+        <Column showEmptyText={false}>
+          {(step >= Step.Claim) && (
+            isPreclaimed
+              ? <AttestDisplay accountId={accountId} />
+              : <ClaimDisplay
+                accountId={accountId}
                 ethereumAddress={ethereumAddress}
+                ethereumSignature={signature}
+                isOldClaimProcess={isOldClaimProcess}
               />
-            )}
-          </Column>
-        </Columar>
-      </main>
-    );
-  }
-
-  protected isDisabled = (): boolean => {
-    const { accountId, signature } = this.state;
-
-    return !accountId || !signature;
-  }
-
-  protected isUnsigned = (): boolean => true;
-
-  protected submitLabel = (): React.ReactNode => this.props.t('Redeem');
-
-  protected txMethod = (): string => 'claims.claim';
-
-  protected txParams = (): [string | null, EcdsaSignature | null] => {
-    const { accountId, signature } = this.state;
-
-    return [
-      accountId ? accountId.toString() : null,
-      signature || null
-    ];
-  }
-
-  protected onChangeAccount = (accountId: string | null): void => {
-    this.setState(({ step }: State): Pick<State, never> => {
-      return {
-        ...(
-          step > Step.Account
-            ? this.defaultState
-            : {}
-        ),
-        accountId
-      };
-    });
-  }
-
-  protected onChangeSignature = (event: React.SyntheticEvent<Element>): void => {
-    const { value: signatureJson } = event.target as HTMLInputElement;
-
-    this.setState(({ step }: State): Pick<State, never> => ({
-      ...(
-        step > Step.Sign
-          ? { step: Step.Sign }
-          : {}
-      ),
-      ...recoverFromJSON(signatureJson)
-    }));
-  }
-
-  private onCopy = (): void => {
-    this.setState({ didCopy: true });
-  }
-
-  private setStep = (step: Step): () => void =>
-    (): void => {
-      this.setState({ step });
-    }
-}
-
-export default withMulti(
-  ClaimsApp,
-  translate,
-  withApi
-);
+          )}
+        </Column>
+      </Columar>
+    </main>
+  );
+};
+
+export default ClaimsApp;
```

Comment:
```suggestion
export default React.memo(ClaimsApp);
```

Does not make a difference here, but consistency everywhere.